### DISCUSSION TITLE
Access Embedded SQLite database within Zip

### DISCUSSION THREAD

**Discussion by krum110487** [zombiezen/go-sqlite#51] [Q&A]
Hello!

I am working on a project that has self contained websites for preservation.
Along with this there are some cases where there are databases involved where dynamic requests are required.

I would like the ability to access a zip that is NOT compressed, but in store mode. So the bits and bytes are all untouched. I am looking to basically open a file with an offset applied so that it can start at the file and process it from there.

I know this is probably not supported out of the box, but I am just looking for a bit of direction or suggestions at where in the code I should look to try to handle this myself. Any kind of hint would be helpful!

Thank you!

**Comment by zombiezen** [comment]
In general, I'd recommend copying the SQLite database to a temporary file and reading it that way. SQLite mostly operates on the filesystem.

However, it does seem that there is a [`sqlite3_deserialize`](https://www.sqlite.org/capi3ref.html#sqlite3_deserialize) function that could handle this for databases small enough to stay in memory (I opened #52 to track adding support for that). Another potential API that could help here is the [virtual file system](https://www.sqlite.org/vfs.html), but it's a bit more involved to implement and I'm not sure I want to commit to that API surface.
